### PR TITLE
chore: babel 7

### PR DESCRIPTION
**Pull request by tuncaulubilge**
Babel 7 version bump, together with couple more dependencies that needs to be bumped for babel 7 compatibility. This PR will enable us to bump react & react-native & react-native-web versions, but that'll be done on a separate pr:

Following the guidelines on this issue: #1540

- [x] Update Babel 7 using babel-upgrade tool
- [x] Replace .babelrc with babel.config.js
- [x] Update Storybooks to v4 (storybook/react-native v5 is not out yet)
- [x] Update Webpack
- [x] Update Graphql and apollo libs
- [x] Get the stories working on web
- [x] Get the stories working on native
- [x] Get the tests working
- [x] Replace Haul with Metro
- [x] Integrate with the apps

 --- What's next for React-Native 0.59+ bump? ---

- [ ] Update React & React-Native & React-Native-Web to latest
- [ ] Update Jest & Babel-Jest
- [ ] Remove extra .babelrc's which were added as a temp fix for old versions of Jest
- [ ] Get the stories working
- [ ] Get the tests working
- [ ] Get the stories and tests working with src instead of transpiled dist
- [ ] Integrate with the apps

### CODE REVIEW

File: .storybook.native/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+const path = require("path");
+
+module.exports = (baseConfig, env, defaultConfig) => {
+  baseConfig.entry = path.resolve(__dirname, "./index.js");
+  baseConfig.resolve = {
+    ...baseConfig.resolve,
+    extensions: [".android.js", ".ios.js", ".native.js", ".js", ".webpack.js", ".mjs", ".js"],
```
Comment (ValGeorgiev) [inline .storybook.native/webpack.config.js]: do we need 2 times ".js"

File: ios-app/.babelrc [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
   "plugins": ["add-react-displayname",
-    [
-      "module-resolver",
-      {
-        "alias": {
-          "react-dom/server": "../node_modules/react-apollo/react-apollo.browser.umd.js"
-        }
-      }
-    ]]
+    // [
```
Comment (nehasri89) [inline ios-app/.babelrc]: remove the comments

File: babel.config.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    plugins: [
+      // [
+      //   "module-resolver",
+      //   {
+      //     alias: {
+      //       "react-dom/server":
+      //         "./node_modules/react-apollo/react-apollo.browser.umd.js"
+      //     }
+      //   }
+      // ],
```
Comment (ValGeorgiev) [inline babel.config.js]: comments

File: ios-app/.babelrc [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
+    //   "module-resolver",
+    //   {
+    //     "alias": {
+    //       "react-dom/server": "../node_modules/react-apollo/react-apollo.browser.umd.js"
+    //     }
+    //   }
+    // ]
```
Comment (ValGeorgiev) [inline ios-app/.babelrc]: comments

**Comment by ValGeorgiev** [comment]
Great job!!! I will do another review, when the conflicts are resolved.

File: .storybook.native/webpack.config.js (hunk 1 above)
Comment (tuncaulubilge) [reply .storybook.native/webpack.config.js]: Oh, nice catch :+1:

File: ios-app/package.json [hunk 5]
```diff
@@ -1,28 +1,27 @@
...
-    "start": "haul start --platform ios",
+    "start": "react-native start",
     "fmt": "prettier --write '**/*.*'",
     "prettier:diff": "prettier --list-different '**/*.*'",
     "lint": "eslint . && yarn prettier:diff"
   },
   "dependencies": {
-    "@times-components/pages": "2.0.8",
-    "prop-types": "15.6.2",
+    "@times-components/pages": "1.23.3",
```
Comment (jmars) [inline ios-app/package.json]: pages downgraded?

Comment (tuncaulubilge) [reply ios-app/package.json]: 🤦‍♂ broke lots of versions on the rebase, fixing it atm

File: .storybook.native/webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  baseConfig.resolve = {
+    ...baseConfig.resolve,
+    extensions: [".android.js", ".ios.js", ".native.js", ".js", ".webpack.js", ".mjs"],
+    mainFields: ["devModule", "dev", "react-native", "browser", "module", "main"],
+  };
+
+  baseConfig.module.rules[0].test = /\.m?jsx?$/
+  baseConfig.module.rules[0].query.presets = ["module:metro-react-native-babel-preset"]
+  baseConfig.module.rules[0].query.plugins = []
+  baseConfig.module.rules[0].exclude = /node_modules\/(?!react|@expo|svgs|pretty-format|haul|metro)/;
```
Comment (ValGeorgiev) [inline .storybook.native/webpack.config.js]: probably we don't need `haul` here

File: .storybook/config.js [hunk 7]
```diff
@@ -1,29 +1,26 @@
...
-
-
-addDecorator((story, context) => withInfo('')(story)(context));
-addDecorator(withKnobs);
+}));
 
-const loadStories = () => req.keys().filter(k => k.indexOf("brightcove-video") === -1).forEach(filename => req(filename));
+const loadStories = () => {
+  req.keys().filter(k => k.indexOf("brightcove-video") === -1).forEach(filename => req(filename));
+};
```
Comment (ValGeorgiev) [inline .storybook/config.js]: hmm i think we are missing return statement here? loadStories now return `undefined`

File: packages/ad/package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@times-components/ad",
-  "version": "2.5.35",
+  "version": "2.5.30",
```
Comment (jmars) [inline packages/ad/package.json]: Looks like something bad is happening with package versions

File: .storybook/config.js (hunk 7 above)
Comment (tuncaulubilge) [reply .storybook/config.js]: It doesn't really need to return something, but I'll revert this :+1:

File: packages/ad/package.json (hunk 8 above)
Comment (tuncaulubilge) [reply packages/ad/package.json]: Should be okay now :+1:

**Pull request merged**